VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitVideoFragment.java

```diff
@@ -288,8 +284,10 @@ public void onNegativeClicked() {
                 MediaConsentUtils.consentToMediaPlayback(getActivity(), dialogCallback,
                         environment.getConfig());
             } else {
-                //Video is downloaded. Hence play
-                startOnlinePlay(entry);
+                if (playerFragment != null && !playerFragment.isFrozen()) {
```

Comment:
This could be converted into a `else if` I guess. Also, unless `setUserVisibleHint()` can be called before `onActivityCreated()`, `playerFragment` should always be initialized, so there is no need for the null check. It's performed in a lot of other places too though, so they can all be removed or it's fine as it is I guess